src/types.ts

```diff
@@ -67,7 +67,7 @@ export interface MapProps {
   tileComponent?: TileComponent
 }
 
-export type TileComponent = (props: TileComponentProps) => JSX.Element
+export type TileComponent = (props: TileComponentProps) => ReactNode
```

Comment:
Thanks! 

Though...

```suggestion
export type TileComponent = (props: TileComponentProps) => ReactElement
```

I believe this should be a `ReactElement` instead, as it'll never be a string or any of the other alternatives in `ReactNode`.